jax/_src/scipy/stats/multinomial.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import scipy.stats as osp_stats
+from jax import lax
+from jax._src.numpy import lax_numpy as jnp
+from jax._src.numpy.util import _wraps, _promote_args_inexact
+from jax._src.scipy.special import gammaln, xlogy
+
+@_wraps(osp_stats.multinomial.logpmf, update_doc=False)
+def logpmf(x, n, p):
+  """JAX implementation of scipy.stats.multinomial.logpmf."""
+  x = jnp.array(x.astype(int))
+  p = _promote_args_inexact("multinomial.logpmf", p)[0]
+  logprobs = gammaln(n + 1) + jnp.sum(xlogy(x, p) - gammaln(x + 1), axis=-1)
```

Comment:
This is going to cause strict type promotion errors because `x` and `p` have different types. We should convert `x` to the type of `p` before attempting operations between them.

Reply:
Yes! Thank you for clarifying sir :) 
I was going back and forth with this.

Reply:
@jakevdp Is promoting different from casting here? I chose this path because `_promote_args_inexact` seems to convert all of the dtypes to a subtype of `floating`. Is there a possibility for x being `float32` and p being `float64`?

Reply:
If you run the tests locally you should see the issue: promoting here is implicit, and will cause test failures due to the `jax_numpy_dtype_promotion=strict` flag used by the test suite. Here we expect promotion because `p` is generally a float, and `x` is required to be an int.